standard/variables.md

```diff
@@ -987,3 +987,201 @@ variable_reference
+- *caller_scope*: Member fields of a `class` type and `ref` parameters have *ref_safe_scope* of *caller_scope*. A variable with *ref_safe_scope* of *caller_scope* can be the referent of a reference return. A variable that can be the referent of a *reference-return* is *safe-to-ref-return*.
+
+These values form a nesting relationship from narrowest (*block*) to widest (*caller_scope*).
+
+> *Example*: The following code shows examples of the different *ref_safe_scope*s. The declarations show the *ref_safe_scope* for a referent to be the initializing expression for a `ref` variable. The examples show the *ref_safe_scope* for a reference return:
+>
+> ```csharp
+> public class C
+> {
+>     private int[] arr = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 };
+> 
+>     public ref int M1(ref int r1)
+>     {
+>         return ref r1; // r1 is safe to ref return
+>     }
+>
+>     public ref int M2(int v1)
+>     {
+>         return ref v1; // error: v1 isn't safe to ref return
+>     }
+>
+>     public ref int M3()
+>     {
+>         int v2 = 5;
+> 
+>         return ref arr[v2]; // arr[v2] is safe to ref return
+>     }
+> 
+>     public void M4(int p) 
+>     {
+>         int v3 = 6;
+>         ref int r2 = ref p; // scope of r2 is block, ref safe scope of p is method
+>         ref int r3 = ref v3; // scope of r3 is block, ref safe scope of v3 is block
+>         ref int r4 = ref arr[v3]; // scope of r4 is block, ref safe scope of arr[v3] is caller method
+>     }
+> }
+> ```
+>
+> *end example.*
+
+#### §ref-span-safety-locals Local variable ref safe scope
+
+For a local variable `v`:
+
+- If `v` is a reference variable, its *ref_safe_scope* is the same as the *ref_safe_scope* of its initializing expression.
+- Otherwise its *ref_safe_scope* is the scope in which it was declared.
+
+#### §ref-span-safety-parameters Parameter ref safe scope
+
+For a formal parameter `p`:
+
+- If `p` is a `ref`, or `in` parameter, its *ref_safe_scope* is the *caller_scope*. It is *safe_to_ref_return*. If `p` is an `in` parameter, it can't be returned as a writable `ref` but can be returned as `ref readonly`.
+- If `p` is an `out` parameter, its *ref_safe_scope* is the *function_member*. It isn't *safe_to_ref_return*.
+- Otherwise, if `p` is the `this` parameter of a struct type, its *ref_safe_scope* is the *function_member*.
+- Otherwise, the parameter is a value parameter, and its *ref_safe_scope* is the *function_member*. It isn't *safe_to_ref_return*.
+
+#### §ref-span-safety-field-reference Field ref safe scope
+
+For a variable designating a reference to a field, `e.F`:
+
+- If `e` is of a reference type, its *ref_safe_scope* is the *caller_scope*.
+- Otherwise, if `e` is of a value type, its *ref_safe_scope* is the same as the *ref_safe_scope* of `e`.
+
+#### §ref-span-safety-operators Operators
+
+The application of a user-defined operator is treated as a method invocation.
```

Comment:
In meeting: can we remove this line entirely? (User-defined operators can't return by ref or have ref parameters.)